src/DynamoDbModel.php

```diff
@@ -86,11 +86,33 @@ protected function setupDynamoDb()
 
     public function save(array $options = [])
     {
-        if (!$this->getKey()) {
-            $this->fireModelEvent('creating');
+        $create = !$this->exists;
+        if ($create && $this->fireModelEvent('creating')  === false) {
+            return false;
         }
+        if (!$create && $this->fireModelEvent('updating') === false) {
+            return false;
+        }
+
+        if ($this->fireModelEvent('saving') === false) {
+            return false;
+        }
+
+        $saved = $this->newQuery()->save();
 
-        return $this->newQuery()->save();
+        if (!$saved) {
+            return $saved;
+        }
+
+        $this->exists = true;
+        if ($create) {
+            $this->wasRecentlyCreated = true;
+            $this->fireModelEvent('created');
+        } else {
+            $this->fireModelEvent('updated');
+        }
```

Comment:
```
$this->fireModelEvent($create ? 'created' : 'updated');
```
wouldn't fire both at the same time. It's either `created` or `updated`, which is the same logic as what you have. Unless I misunderstand you.